crates/iceberg/src/io.rs

```diff
@@ -240,9 +241,9 @@ impl InputFile {
 }
 
 /// Trait for writing file.
-pub trait FileWrite: AsyncWrite {}
+pub trait FileWrite: AsyncWrite + TokioAsyncWrite + Send + Unpin {}
```

Comment:
It's quite weird that `FileWrite` requires to have both `AsyncWrite` and `TokioAsyncWrite`. It increases the burden for the downstream to implement.

Did I missed something here?

Reply:
I added TokioAsyncWrite because the parquet writer accepct a tokio writer. Can we remove the futures::AsyncWrite? Since for now, we don't need it.

Reply:
> Can we remove the futures::AsyncWrite? Since for now, we don't need it.

That's fine to remove `futures::AsyncWrite` here if not needed.

But I want to mention that using `AsyncWrite` is more friendly for users who don't want to depend on `TokioAsyncWrite` and there are `async_compat` crate to transform a `futures::AsyncWrite` to `TokioAsyncWrite`.

Maybe we can add our own trait to support both of them? I believe this is another issue and should not be part of this PR.